hal/src/rtl872x/rtc_hal.cpp

```diff
@@ -331,12 +333,14 @@ RealtekRtc rtcInstance;
 } // anonymous
 
 void hal_rtc_init(void) {
-    rtcInstance.init();
-    struct timeval tv = {
-        .tv_sec = UNIX_TIME_20000101000000,
-        .tv_usec = 0
-    };
-    rtcInstance.setTime(&tv);
+    if (HAL_READ32(SYSTEM_CTRL_BASE_LP, REG_LP_DSLP_INFO_SW) == false) {
```

Comment:
I would maybe also additionally double check that `timeInfo_` is valid.

Reply:
To check the validity of `timeInfo_ `, we need to call `mktime()`, but it takes too long to execute at this point. See the start timestamp:
>0000010171 [system] INFO: Device 0a10aced202194944a04072c started

Reply:
The root cause is `timeInfo_ ` marked as retained variable. Both system retained and static data are resided in interal SRAM, I have no idea why accessing the retained data takes this long time.

Reply:
The reason why `mktime()` takes so long is becuase the `timeInfo_` is not initialized (i.e. restore from the backup flash page) after the device wakes up from hibernate mode.